nodedb.go

```diff
@@ -385,14 +457,56 @@ func (ndb *nodeDB) deleteVersion(version int64) error {
 	if err != nil {
 		return err
 	}
-	if rootKey == nil || GetNodeKey(rootKey).version < version {
-		if err := ndb.batch.Delete(ndb.nodeKey(GetRootKey(version))); err != nil {
+
+	literalRootKey := GetRootKey(version)
+	if rootKey == nil || !bytes.Equal(rootKey, literalRootKey) {
+		// if the root key is not matched with the literal root key, it means the given root
+		// is a reference root to the previous version.
+		if err := ndb.deleteFromPruning(ndb.nodeKey(literalRootKey)); err != nil {
```

Comment:
This use of batch deletes is unsafe. For example, we let's say we batch delete the root key for the target version. If the main routine saves a tree, it will write the batch to disk. Then `traverseOrphans` will fail because it assumes that the target version is still on disk